src/main/java/com/restaurante/controllers/CompanyController.java

```diff
@@ -14,22 +20,28 @@ public class CompanyController {
     private final CompanyServices companyServices;
 
     @PostMapping
-    public ResponseEntity<Object> createCompany(@RequestBody CompanyDto companyDto) {
-        return companyServices.createCompany(companyDto);
+    public ResponseEntity<Object> createCompany(@RequestBody CompanyDto companyDto)
+            throws CompanyExistException, InternalServerErrorException {
+        companyServices.createCompany(companyDto);
+        return ResponseEntity.status(HttpStatus.CREATED).body("Company created.");
```

Comment:
Quando você vai retornar um status code **201**, é esperado que seja retornado um _cabeçalho_(**Headers**) contendo o novo endereço que voce inseriu:
Exemplo que ficaria no **headers**: _http://localhost:8080/company/3_
Uma maneira de fazer isso, é utilizando o **URI** e utilizando um metodo ._created()_ do proprio _ResponseEntity_

```suggestion
        URI uri = ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}").buildAndExpand(obj.getId()).toUri();
        return ResponseEntity.created(uri).body("Company created.");
```

Reply:
Não está buscando o id cadastrado para completar a URI

Reply:
No service, faz retornar o objeto, ai no controller voce acrescente a URI, mas não precisa disponibilizar o objeto